packages/smol/components/sections/Revoke/AllowanceRow.tsx

```diff
@@ -0,0 +1,95 @@
+import type {TExpandedAllowance, TTokenAllowance} from '@lib/types/Revoke';
+
+type TAllowanceRowProps = {
+	allowance: TExpandedAllowance;
+	revoke: (tokenToRevoke: TTokenAllowance, spender: TAddress) => void;
+};
+
+export const AllowanceRow = ({allowance, revoke}: TAllowanceRowProps): ReactElement => {
+	const {args, transactionHash} = allowance;
+	const allowanceAmount = useMemo(() => {
+		if (isUnlimited(allowance.args.value as bigint)) {
+			return 'Unlimited';
+		}
+		return getTokenAmount(allowance.decimals, allowance.args.value as bigint);
+	}, [allowance]);
+
+	return (
+		<tr key={transactionHash}>
+			<td className={'rounded-l-lg border-y border-l border-neutral-400 p-6'}>
+				<div className={'flex'}>
+					<div>
+						<ImageWithFallback
+							alt={allowance.symbol ?? ''}
+							unoptimized
+							src={`${process.env.SMOL_ASSETS_URL}/token/${allowance.chainID}/${allowance.address}/logo-32.png`}
+							altSrc={`${process.env.SMOL_ASSETS_URL}/token/${allowance.chainID}/${allowance.address}/logo-32.png`}
+							quality={90}
+							width={40}
+							height={40}
+						/>
+					</div>
+					<div className={'ml-4 flex flex-col'}>
+						<div className={'text-base font-bold'}>{allowance.symbol}</div>
+
+						<button
+							className={
+								'z-10 flex w-full cursor-copy items-center justify-end font-light text-neutral-600'
+							}
+							onClick={e => {
+								e.stopPropagation();
+								navigator.clipboard.writeText(toAddress(allowance.address));
+								toast.success(`Address copied to clipboard: ${toAddress(allowance.address)}`);
+							}}>
+							<p className={'mb-[-2px] mr-1 text-xs hover:underline'}>
+								{truncateHex(allowance.address, 5)}
+							</p>
+						</button>
+					</div>
+				</div>
+			</td>
+			<td className={'p-y-6 max-w-[80px] border-y border-neutral-400'}>
+				<div className={'flex w-full justify-end'}>
+					<p className={'h-full truncate text-right text-base leading-6'}>{allowanceAmount}</p>
+				</div>
+			</td>
+			<td className={'max-w-32 border-y border-neutral-400 p-6'}>
+				<div className={'flex flex-col text-right'}>
+					<div className={'flex items-center justify-end font-light text-neutral-600'}>
+						<div className={'grid w-full'}>
+							<button
+								className={
+									'z-10 flex w-full cursor-copy items-center justify-end font-light text-neutral-600'
+								}
+								onClick={e => {
+									e.stopPropagation();
+									navigator.clipboard.writeText(toAddress(args.sender));
+									toast.success(`Address copied to clipboard: ${toAddress(args.sender)}`);
+								}}>
+								<p className={'mb-[-2px] text-xs hover:underline'}> {truncateHex(args.sender, 5)}</p>
+							</button>
+						</div>
+					</div>
+				</div>
+			</td>
+			<td className={'w-32 rounded-r-lg border-y border-r border-neutral-400 p-3'}>
+				<Button
+					onClick={() =>
+						revoke({address: allowance.address, name: allowance.symbol ?? ''}, allowance.args.sender)
```

Comment:
name should be extra here

Reply:
Name is required here to be able to display it in Success modal after revoke